andela/travel_tool_back
# #160233501 Assigning manager role to user

#### What does this PR do?
Modifies the endpoint for getting users with a particular role to include the users' names

#### Description of Task to be completed?
- Include the full name of the user in the model association between roles and users
- Fetch users with the role of manager

#### How should this be manually tested?
- Pull the branch
- Enter into the project directory - `cd travel_tool_back`
- Start the app using `make start`
- Open `Postman`
- Create a new user by sending a `POST` request to the endpoint 
`http://localhost:5000/api/v1/user`  with your `email`, `fullName` and `userId` field
- Set the default super admin by adding your andela email to the dotenv file. Check env.example file for sample. send a `PUT` request to the endpoint `http://localhost:5000/api/v1/user/admin` . This will allow you assign a manger role to yourself.
 - Update your role to manger by sending a `PUT` request to the endpoint
`http://localhost:5000/api/v1/user/role/update`  and  fields with `email` (your andela email address), roleName for the role  which is `manager`
- To get users with the manager role, send a `GET` request to the endpoint `http://localhost:5000/api/v1/user/roles/53019` , you should see your name and your email as one of the users listed with the role of manager.
 

#### Any background context you want to provide?
- Run yarn db:seed to seed the default roles to the data base
- Only the Super admin can change user role and add new roles to the database. The default role when a new user sign up is `Requester`
- All the endpoint requires authentication. To include token in the `authorization` header:
- Login to the [travela app](https://travela-staging.andela.com)
- After you are logged in, visit the developer console in your browser and open `Application`, you should see your `jwt-token` there.
- Copy the token and add it to the `authorization` header when making a request in postman.
- The set default super admin endpoint will only make the person a super admin if the user email address in the dotenv file and the one from the jwt token are the same and the user already exist in the database


#### Any background context you want to provide?
None

#### What are the relevant pivotal tracker stories?
[#160233501](https://www.pivotaltracker.com/story/show/160233501)

#### Screenshots (if appropriate)
<img width="1440" alt="screen shot 2018-09-03 at 7 51 30 pm" src="https://user-images.githubusercontent.com/25195857/44999477-7021ba00-afb5-11e8-9521-a12955cd6baf.png">


#### Questions:
None
